packages/asset/contracts/AssetReveal.sol

```diff
@@ -61,12 +69,13 @@ contract AssetReveal is
         IAsset.AssetData memory data = tokenId.getData();
         require(!data.revealed, "Asset is already revealed");
         assetContract.burnFrom(_msgSender(), tokenId, amount);
+        nonce[_msgSender()]++;
         emit AssetRevealBurn(
             _msgSender(),
             tokenId,
+            nonce[_msgSender()],
```

Comment:
I don't understand the need of this nonce variable
Is it for the backend to be able to pass the variable `signatureNonce` to the mint ?

Reply:
And if yes, why not just remove the nonce from the burn & increment the nonce during the mint ? `prevTokenId` should be enough to mint

Reply:
Yeah, so the nonce was emitted in the burn event to let the backend know what nonce to use when generating a signature, but your suggestion is valid, let me run it through my mind and try to see if I can make it work.
Might actually be better, thanks @mvanmeerbeck !